packages/amplify-graphql-relational-transformer/src/__tests__/amplify-graphql-has-one-transformer.test.ts

```diff
@@ -305,17 +307,19 @@ test('creates has one relationship with implicit fields', () => {
 
   const createInput = schema.definitions.find((def: any) => def.name && def.name.value === 'CreateTestInput') as any;
   expect(createInput).toBeDefined();
-  expect(createInput.fields.length).toEqual(3);
+  expect(createInput.fields.length).toEqual(4);
   expect(createInput.fields.find((f: any) => f.name.value === 'id')).toBeDefined();
   expect(createInput.fields.find((f: any) => f.name.value === 'email')).toBeDefined();
   expect(createInput.fields.find((f: any) => f.name.value === 'testOtherHalfId')).toBeDefined();
+  expect(createInput.fields.find((f: any) => f.name.value === 'testOtherHalfEmail')).toBeDefined();
```

Comment:
Can we update the pre-processing logic and add some pre-processing unit tests in around this? Planning to start work on enabling that next week in the transformer. Specifically the changes would be in the `mutateSchema` methods in the relational transformers